cmake/cudawrappers-inline-local-includes.cmake

```
@@ -16,14 +24,35 @@ function(inline_local_includes input_file output_file root_dir)
     if(NOT INCLUDE_PATHS STREQUAL "")
       list(SORT INCLUDE_PATHS ORDER DESCENDING)
       list(GET INCLUDE_PATHS 0 include_PATH)
-      list(APPEND include_files ${include_PATH})
-      file(READ ${include_PATH} include_contents)
-      string(REPLACE "${include_line}" "${include_contents}"
-                     input_file_contents "${input_file_contents}"
-      )
+      list(FIND ALREADY_INCLUDED ${INCLUDE_PATHS} found_index)
+      if(found_index EQUAL -1)
+        list(APPEND ALREADY_INCLUDED ${include_PATH})
+        set(include_contents "")
+        inline_local_includes(${include_PATH} include_contents ${root_dir})
+        # Replace the include line with the include file contents
+        string(REPLACE "${include_line}" "${include_contents}"
+                       input_file_contents "${input_file_contents}"
+        )
+      else()
+        # Remove the include line
+        string(REPLACE "${include_line}" "" input_file_contents
+                       "${input_file_contents}"
+        )
+      endif()
```

Not sure if it's good for readability, but this could be shortened a bit by setting `include_contents` to an empty string before the if statement:

```suggestion
      set(include_contents "")
      if(found_index EQUAL -1)
        list(APPEND ALREADY_INCLUDED ${include_PATH})
        inline_local_includes(${include_PATH} include_contents ${root_dir})
      endif()
      # Replace the include line with the include file contents or
      # remove it if the file was already included
      string(REPLACE "${include_line}" "${include_contents}"
                     input_file_contents "${input_file_contents}"
      )
```

---

I like it! :+1: